File: examples/01-foundations/go/core_primitives.go
package foundations

import (
	"context"

	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
	"go.temporal.io/sdk/workflow"
)

// DAG -> Workflow: durable orchestration state and decisions.
func LoanWorkflow(ctx workflow.Context, applicationID string) (string, error) {
	return "", nil
}

// Operator -> Activity: unreliable work with retries, timeouts, side effects.
// In Go, Activities are plain functions whose first arg is context.Context.
func PullCreditScore(ctx context.Context, applicationID string) (int, error) {
	return 0, nil
}

func NotifyApplicant(ctx context.Context, applicationID, decision string) error {
	return nil
}

// Executor -> Worker: a long-running process polling a Task Queue.
func start(c client.Client) error {
	w := worker.New(c, "loan-decisions", worker.Options{})
	w.RegisterWorkflow(LoanWorkflow)
	w.RegisterActivity(PullCreditScore)
	w.RegisterActivity(NotifyApplicant)
	return w.Run(worker.InterruptCh())
}

File: examples/01-foundations/python/core_primitives.py
from temporalio import activity, workflow
from temporalio.client import Client
from temporalio.worker import Worker


# DAG -> Workflow: durable orchestration state and decisions.
@workflow.defn
class LoanWorkflow:
    @workflow.run
    async def apply(self, application_id: str) -> str: ...


# Operator -> Activity: unreliable work with retries, timeouts, and side effects.
# In Python, Activities are plain functions marked with @activity.defn.
@activity.defn
async def pull_credit_score(application_id: str) -> int: ...


@activity.defn
async def notify_applicant(application_id: str, decision: str) -> None: ...


# Executor -> Worker: a long-running process polling a Task Queue.
async def start(client: Client) -> None:
    worker = Worker(
        client,
        task_queue="loan-decisions",
        workflows=[LoanWorkflow],
        activities=[pull_credit_score, notify_applicant],
    )
    await worker.run()
